[PL-Doc] Platform Release Notes 1.99.x (#11083)
harness/developer-hub: release-notes/platform.md

## July 2025

### Version 1.99.x <!--July 30, 2025-->

#### Fixed issues

- Increased the item limit in scrollable lists beyond 100 to enhance usability and avoid display limitations. [PL-63813]

> #### New Feature and Enhancement

Reviewer: @ivikramsahu this heading used to be "New features and enhancements". We have recently changed it to "New Feature and Enhancement". Please ensure such errors are not done. Also, please fix it at all the recent places.

## July 2025

### Version 1.99.x <!--July 30, 2025-->

#### Fixed issues

- Increased the item limit in scrollable lists beyond 100 to enhance usability and avoid display limitations. [PL-63813]

#### New Feature and Enhancement

- Introduced [Secret Manager template reconciliation](/docs/platform/secrets/secrets-management/reconcilation-of-secret-manager-template) to automatically review and update linked Secret Managers and Secrets when templates are modified, preventing misconfigurations. [PL-61310]

> - Added [Default Notification Templates](/docs/platform/notifications/default-notification-template) that automatically apply when no custom template is selected, ensuring consistent notifications across scopes. [PL-61479]

Reviewer: ```suggestion
- Launched [Default Notification Templates](/docs/platform/notifications/default-notification-template) that automatically apply when no notification template is selected in a notification rule, providing the capability to define the notification template centrally. [PL-61479]
```

> - Introduced [Centralised Delegate Notifications](/docs/platform/notifications/centralised-notification) for proactive delegate monitoring with immediate alerts for disconnection, expiration, and connectivity issues. [PL-31873]

Reviewer: ```suggestion
- Introduced [Centralised Delegate Notifications](/docs/platform/notifications/centralised-notification) for proactive delegate monitoring with immediate alerts for disconnection, expiration, and near-expiration alerts. [PL-31873]
```

> - AWS KMS connector now supports [plain text and encrypted ARNs](/docs/platform/secrets/secrets-management/add-an-aws-kms-secrets-manager) across all authentication methods for enhanced security. [PL-62434]

Reviewer: ```suggestion
- AWS KMS connector now supports adding [ARN as plain text](/docs/platform/secrets/secrets-management/add-an-aws-kms-secrets-manager) across all authentication methods. [PL-62434]
```

- Enhanced database operations during delegate registration to prevent duplicate entries in `delegateConfig`. [PL-64095]
- Improved visibility on the service accounts' details page at a higher scope; users can now see details of service accounts that are inherited at lower scopes. [PL-63634]

> - A new API has been introduced to delete the delegate token. [PL-63386]

Reviewer: @ivikramsahu this is not just limited to API but has UI changes as well. Let's update the Release notes and docs as well.
